pkg/kn/commands/source/apiserver/flags.go

```diff
@@ -40,47 +42,99 @@ type APIServerSourceUpdateFlags struct {
 	Resources          []string
 }
 
-type resourceSpec struct {
-	kind         string
-	apiVersion   string
-	isController bool
-}
-
-// GetAPIServerResourceArray is to return an array of ApiServerResource from a string. A sample is Event:v1:true,Pod:v2:false
-func (f *APIServerSourceUpdateFlags) GetAPIServerResourceArray() (*[]v1alpha1.ApiServerResource, error) {
+// getAPIServerResourceArray is to construct an array of resources.
+func (f *APIServerSourceUpdateFlags) getAPIServerResourceArray() ([]v1alpha1.ApiServerResource, error) {
 	var resourceList []v1alpha1.ApiServerResource
 	for _, r := range f.Resources {
 		resourceSpec, err := getValidResource(r)
 		if err != nil {
 			return nil, err
 		}
-		resourceRef := v1alpha1.ApiServerResource{
-			APIVersion: resourceSpec.apiVersion,
-			Kind:       resourceSpec.kind,
-			Controller: resourceSpec.isController,
+		resourceList = append(resourceList, *resourceSpec)
+	}
+	return resourceList, nil
+}
+
+// updateExistingAPIServerResourceArray is to update an array of resources.
+func (f *APIServerSourceUpdateFlags) updateExistingAPIServerResourceArray(existing []v1alpha1.ApiServerResource) ([]v1alpha1.ApiServerResource, error) {
```

Comment:
I would like to move this method to parsing_helper.go to provide a common method to update existing array.
My question is which type (or key word) I should use to describe a general array here ?

I tried to use []interface{} but I get an error:
```
cannot use existing (type []"knative.dev/eventing/pkg/apis/sources/v1alpha1".ApiServerResource) as type []interface {} in argument to util.UpdateArray
```

Reply:
That's not easily possible, see https://golang.org/doc/faq#convert_slice_of_interface

Reply:
What you could do is to introduce a simple utility method which converts an `[]ApiServerResource` to a `[]interface{}` and feed that into the generic method like in 

```golang
fromISlice(UpdateArray(toISlice(existing), toISliceP(added), toISliceP(removed)))
```

and do the conversion between ApiServerSource and interfaces slices before and after calling the function. But I don't think its worth the hassle.

Reply:
oh, ok. Then I keep as it is. 
Please review the latest version. 
Thank you.